x-pack/plugins/actions/common/validate_email_addresses.ts

```diff
@@ -80,32 +73,33 @@ function validateEmailAddress(
   }
 }
 
-function validateEmailAddress_(allowedDomains: string[], address: string): ValidatedEmail {
+function validateEmailAddress_(allowedDomains: string[] | null, address: string): ValidatedEmail {
   const emailAddresses = parseAddressList(address);
   if (emailAddresses == null) {
     return { address, valid: false, reason: InvalidEmailReason.invalid };
   }
 
-  const allowedDomainsSet = new Set(allowedDomains);
+  if (allowedDomains !== null) {
+    const allowedDomainsSet = new Set(allowedDomains);
 
-  for (const emailAddress of emailAddresses) {
-    let domains: string[] = [];
+    for (const emailAddress of emailAddresses) {
+      let domains: string[] = [];
 
-    if (emailAddress.type === 'group') {
-      domains = emailAddress.addresses.map((groupAddress) => groupAddress.domain);
-    } else if (emailAddress.type === 'mailbox') {
-      domains = [emailAddress.domain];
-    } else {
-      return { address, valid: false, reason: InvalidEmailReason.invalid };
-    }
+      if (emailAddress.type === 'group') {
+        domains = emailAddress.addresses.map((groupAddress) => groupAddress.domain);
+      } else if (emailAddress.type === 'mailbox') {
+        domains = [emailAddress.domain];
+      } else {
+        return { address, valid: false, reason: InvalidEmailReason.invalid };
+      }
 
-    for (const domain of domains) {
-      if (!allowedDomainsSet.has(domain)) {
-        return { address, valid: false, reason: InvalidEmailReason.notAllowed };
+      for (const domain of domains) {
+        if (!allowedDomainsSet.has(domain)) {
+          return { address, valid: false, reason: InvalidEmailReason.notAllowed };
+        }
       }
     }
   }
```

Comment:
Do we need to do the check for `emailAddress.type === 'group'` or `emailAddress.type === 'mailbox'` or else return invalid even if allowedDomains is null?

Reply:
The whole block is in an if block (line 82).
We don't check for `emailAddress.type` or loop through domains if allowedDomains is `null`.